FLT/DivisionAlgebra/ModuleTopProd.lean

```diff
@@ -0,0 +1,355 @@
+import Mathlib.Topology.Algebra.Module.ModuleTopology
+
+namespace ModuleTopIsProd
+
+variable (R S : Type*) [Ring R] [Ring S] [TopologicalSpace R] [TopologicalSpace S] (M N : Type*)
+  [AddCommMonoid M] [AddCommMonoid N] [Module R M] [Module S N] [TopologicalSpace M]
+  [IsModuleTopology R M] [TopologicalSpace N] [IsModuleTopology S N]
+
+instance : HSMul (R × S) (M × N) (M × N) where
+  hSMul a b := (a.1 • b.1, a.2 • b.2) -- why is this needed...
+
+-- maybe there is an easier way to do this without being explicit?
+instance : Module (Prod R S) (Prod M N) where
+  smul a b := (a.1 • b.1, a.2 • b.2)
+  one_smul a := by
+    have : (1 : (R × S)) • a= ((1 : R × S).1 • a.1, (1 : R × S).2 • a.2) := by
+      rfl
+    aesop
+  mul_smul a b c := by
+    have : (a * b) • c = ((a * b).1 • c.1, (a * b).2 • c.2) := by
+      rfl
+    simp_rw [this, Prod.fst_mul, Prod.snd_mul, mul_smul]
+    rfl
+  smul_zero a := by
+    have : a • (0 : M × N) = (a.1 • 0, a.2 • 0) := by
+      rfl
+    aesop
+  zero_smul a := by
+    have : (0 : R × S) • a = ((0 : R) • a.1, (0 : S) • a.2) := by
+      rfl
+    aesop
+  smul_add a b c:= by
+    have : a • (b + c) = (a.1 • (b + c).1, a.2 • (b + c).2) := by
+      rfl
+    aesop
+  add_smul a b c := by
+    have : (a + b) • c = ((a + b).1 • c.1, (a + b).2 • c.2) := by
+      rfl
+    simp_rw [this, Prod.fst_add, Prod.snd_add, add_smul]
+    rfl
+
+-- not sure why I need to add everything in here...
+lemma continuous_add (R S : Type*) [Ring R] [Ring S] [TopologicalSpace R] [TopologicalSpace S]
+    (M N : Type*) [AddCommMonoid M] [AddCommMonoid N] [Module R M] [Module S N] [TopologicalSpace M]
+    [IsModuleTopology R M] [TopologicalSpace N] [IsModuleTopology S N] :
+    ContinuousAdd (Prod M N) := by
+  have a : ContinuousAdd M := IsModuleTopology.toContinuousAdd R M
+  have b : ContinuousAdd N := IsModuleTopology.toContinuousAdd S N
+  exact Prod.continuousAdd
+
+/-- Permutation map of `(R × S) × (M × N) → (R × M) × (S × N)`. -/
+def permutation_map : (R × S) × (M × N) → (R × M) × (S × N) :=
+  fun a => ((a.1.1, a.2.1), (a.1.2, a.2.2))
```

Comment:
Surely we have this already? But I don't know what it's called. does `exact?` find it?